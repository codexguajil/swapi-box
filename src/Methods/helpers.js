import React from 'react';

export const iterateHelper = async (array, path) => {
    // console.log(array)
    if (path === 'people') {
        try {
            let objectValues = array.map(item => {
                return {name: item.name,
                        home: item.homeworld,
                        species: item.species[0]
                }
            })
            return await objectValues
        } catch(error) {
            return error.message
        }
    } else if (path === 'planets') {
        try {
            let objectValues = array.map(item => {
                return {name: item.name,
                        terrain: item.terrain,
                        population: item.population,
                        climate: item.climate,
                        residents: item.residents
                }
            })
            return await objectValues
        } catch(error) {
            return error.message
        }
    } else if (path === 'vehicles') {
        try {
            let objectValues = array.map(item => {
                return {name: item.name,
                        model: item.model,
                        class: item.vehicle_class,
                        passengers: item.passengers
                }
            })
            return await objectValues
        } catch(error) {
            return error.message
        }
    }
    
}