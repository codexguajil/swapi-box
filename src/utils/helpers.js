import { fetchAnything } from './apiCalls';

export const peopleCleaner = (data) => {
  return data.map(person => {
    return {
            name: person.name,
            homeUrl: person.homeworld,
            speciesUrl: person.species[0],
    }
  })
}

export const homeWorldData = async (data) => {
  const promises =  data.map(person => {
    return fetchAnything(person.homeUrl)
  })
  return Promise.all(promises);
}

export const speciesData = async (data) => {
  const promises = data.map(person => {
    return fetchAnything(person.speciesUrl)
  })
  return Promise.all(promises)
}

export const speciesDataCleaner = (data) => {
  return data.map(species => {
    return species.name
  })
}

export const combineData = (people, planets, species) => {
  return people.map((person,i) => {
    return {
      name: person.name,
      homeworld: planets[i].name,
      species: species[i],
      population: planets[i].population
    }
  })
}








//     } else if (path === 'planets') {
//         try {
//             let objectValues = array.map(item => {
//                 return {name: item.name,
//                         terrain: item.terrain,
//                         population: item.population,
//                         climate: item.climate,
//                         residents: item.residents
//                 }
//             })
//             return objectValues
//         } catch(error) {
//             return error.message
//         }
//     } else if (path === 'vehicles') {
//         try {
//             let objectValues = array.map(item => {
//                 return {name: item.name,
//                         model: item.model,
//                         class: item.vehicle_class,
//                         passengers: item.passengers
//                 }
//             })
//             return objectValues
//         } catch(error) {
//             return error.message
//         }
//     }
    
// }