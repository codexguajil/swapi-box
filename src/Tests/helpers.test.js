import React from 'react';
import { iterateHelper } from '../Methods/helpers';
import { shallow } from 'enzyme';

describe("helpers", () => {
    describe("iterateHelper", () => {
        let wrapper;
        let mockPath;
        let array;

        beforeEach(() => {
            array = [
                {
                    name: 'ed',
                    model: 'v2'
                },
                {
                    name: 'voks',
                    model: 'sapien'
                },
                {
                    name: 'thumper',
                    model: 'rainier'
                }
            ]
            mockPath = 'vehicle'
            let testProps = {array, mockPath}
            wrapper = shallow( <iterateHelper {...testProps} />)
        })

        it("should return the correct array of objects based on params", () => {

            expect(wrapper).toMatchSnapshot()
        })
    })
})