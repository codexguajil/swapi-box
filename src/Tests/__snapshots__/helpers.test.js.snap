// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers iterateHelper should return the correct array of objects based on params 1`] = `
<iterateHelper
  array={
    Array [
      Object {
        "model": "v2",
        "name": "ed",
      },
      Object {
        "model": "sapien",
        "name": "voks",
      },
      Object {
        "model": "rainier",
        "name": "thumper",
      },
    ]
  }
  mockPath="vehicle"
/>
`;
